### PR TITLE
Add function to clean up TBObjects

### PR DESCRIPTION
**Pull request by mcmikashi**

### CODE REVIEW

**Comment by JeffAbrahamson** [comment]
I think `update_tb_object_scheduled_date()` should perhaps filter on the current slug.  Two reasons motivate me.  First, imagine that the system starts to be large.  Filtering to current slug would reduce the amount of work that potentially needs to happen on viewing the list.  Second, in the (highly unlikely) event that we have lots of concurrent users, it reduces the chances of conflict.  Although postgresql should isolate adequately to avoid that, but that isolation comes at some cost.

**Comment by JeffAbrahamson** [comment]
That would be `K_SCHEDULE_FOR_DELETION` rather than `K_SCHEDULE_FOR_DELATION`. ;-)

**Comment by JeffAbrahamson** [comment]
I don't think we need to create a new util.py file.  The constant is naturally a property of the TBObject base class.

The two functions operate on slugs (object classes) rather than individual TBObjects, so let's put them in views.py with names that reflect that.  Also, we might encode a bit of context in the function names.

```
def schedule_moribund_slug_instances_for_deletion(tb_object)
def delete_scheduled_moribund_slug_instances(tb_object)
```

Here, `tb_object` _might_ be an actual object.  (Otherwise, call it a class, because it's a big confusing.)  And then you can do the search by saying (I'm totally inventing this and haven't tested it) something like this:

```
tb_object.model.objects.filter(
    slug=tb_object.slug,
    scheduled_for_deletion_date=None).update(...)
```
etc.  Wdyt?

Both of those functions could benefit from docstrings.  As well as the model changes, too.

**Comment by JeffAbrahamson** [comment]
Don't forget to tag the issue #725.

**Comment by Shriukan33** [comment]
Also, do we want to delete all not published items for a slug ? I thought we would like to keep the previously published versions, so an extra condition would be `first_publication_date == None`.

**Comment by JeffAbrahamson** [comment]
> Also, do we want to delete all not published items for a slug ? I thought we would like to keep the previously published versions, so an extra condition would be `first_publication_date == None`.

Correct, we should keep anything that has ever been published and only delete old things that have never been published.

**Comment by Shriukan33** [comment]
The migration file is number 42, while master is already at 43. 
To avoid conflicts on master, please rebase and re-do your migration file :)

**Comment by mcmikashi** [comment]
- Rebase on master
- Add test
- Move the functions on the view

File: transport_nantes/topicblog/forms.py [hunk 1]
```diff
@@ -14,7 +14,8 @@ class Meta:
         model = TopicBlogItem
         # Admins can still edit those values
         exclude = ('item_sort_key', 'servable', 'user',
-                   'publication_date', 'first_publication_date', 'publisher')
+                   'publication_date', 'first_publication_date', 'publisher',
+                   'scheduled_for_deletation_date')
```
Comment (Shriukan33) [inline transport_nantes/topicblog/forms.py:18]: deletation -> Deletion

File: transport_nantes/topicblog/templates/topicblog/topicblogbase_list_one.html [hunk 2]
```diff
@@ -42,6 +46,12 @@
             <br>
             Créé {{ object.date_modified|date:"l d-m-Y à G:i" }} par {{ object.user.email }}
             <br>
+            {% if not object.first_publication_date and not object.publication_date and object.scheduled_for_deletation_date  %}
+                <span class="{% if object.get_days_defore_deletation <= 0 %} text-danger {% endif %}">
```
Comment (Shriukan33) [inline transport_nantes/topicblog/templates/topicblog/topicblogbase_list_one.html:50]: deletation -> Deletion

File: transport_nantes/topicblog/migrations/0047_topicblogemail_scheduled_for_deletation_date_and_more.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# Generated by Django 4.0.4 on 2022-05-23 16:19
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('topicblog', '0046_auto_20220518_1241'),
```
Comment (Shriukan33) [inline transport_nantes/topicblog/migrations/0047_topicblogemail_scheduled_for_deletation_date_and_more.py:9]: We added some more migrations, we're now at 48

**Comment by JeffAbrahamson** [comment]
Thanks!

In addition to the other comments in the review, could you do the following before we merge:
* s/deletation/deletion/;
* tag issue
* fix the trailing white space

**Comment by mcmikashi** [comment]
- Fix spelling
- Reword the commit message
- Remove trailing white space

**Comment by Shriukan33** [comment]
Could you please rebase ? :)

**Comment by mcmikashi** [comment]
- Rebased

**Review by Shriukan33 — APPROVED** [review]
LGTM

**Comment by Shriukan33** [comment]
There are conflicts in rebase now @mcmikashi :/

**Comment by mcmikashi** [comment]
- Rebased

**Comment by Shriukan33** [comment]
@JeffAbrahamson  ?

**Comment by JeffAbrahamson** [comment]
```
localhost tn_beta_django@tn_beta2_db=> SELECT
            first_publication_date, publication_date, 
            date_modified, scheduled_for_deletion_date
        FROM topicblog_topicblogitem
        WHERE slug = 'devenir-benevole';
    first_publication_date    |       publication_date       |         date_modified         | scheduled_for_deletion_date 
------------------------------+------------------------------+-------------------------------+-----------------------------
 (NULL)                       | (NULL)                       | 2022-02-03 17:23:40.703823+00 | (NULL)
 (NULL)                       | (NULL)                       | 2022-02-03 17:23:40.742883+00 | (NULL)
 (NULL)                       | (NULL)                       | 2022-02-03 17:23:40.842589+00 | (NULL)
 (NULL)                       | (NULL)                       | 2022-02-03 17:23:40.925381+00 | (NULL)
 (NULL)                       | (NULL)                       | 2022-02-03 17:23:40.938666+00 | (NULL)
 (NULL)                       | (NULL)                       | 2022-02-03 17:23:41.117461+00 | (NULL)
 (NULL)                       | (NULL)                       | 2022-02-03 17:23:41.132057+00 | (NULL)
 (NULL)                       | (NULL)                       | 2022-05-18 13:22:57.935134+00 | (NULL)
 2022-05-18 13:24:55.42542+00 | 2022-05-18 13:24:55.42542+00 | 2022-05-18 13:24:55.447974+00 | (NULL)
 (NULL)                       | (NULL)                       | 2022-02-03 17:23:40.914582+00 | (NULL)
 (NULL)                       | (NULL)                       | 2022-02-03 17:23:40.928461+00 | (NULL)
 (NULL)                       | (NULL)                       | 2022-02-03 17:23:40.931772+00 | (NULL)
 (NULL)                       | (NULL)                       | 2022-02-03 17:23:40.935062+00 | (NULL)
 (NULL)                       | (NULL)                       | 2022-02-03 17:23:40.942292+00 | (NULL)
 (NULL)                       | (NULL)                       | 2022-02-03 17:23:40.945654+00 | (NULL)
 (NULL)                       | (NULL)                       | 2022-02-03 17:23:40.948656+00 | (NULL)
 (NULL)                       | (NULL)                       | 2022-02-03 17:23:41.464377+00 | (NULL)
 (NULL)                       | (NULL)                       | 2022-02-03 17:23:41.468015+00 | (NULL)
 (NULL)                       | (NULL)                       | 2022-02-03 17:23:41.514054+00 | (NULL)
 (NULL)                       | (NULL)                       | 2022-02-03 17:23:40.90771+00  | (NULL)
(20 rows)

Time: 1.917 ms
localhost tn_beta_django@tn_beta2_db=>
```

**Comment by Shriukan33** [comment]
Did you visit the delete url ?

**Comment by JeffAbrahamson** [comment]
> Did you visit the delete url ? You need to click on the trashcan icon I believe to start the countdown

Hmm, perhaps I should try that. ;-)

**Comment by JeffAbrahamson** [comment]
> > Did you visit the delete url ? You need to click on the trashcan icon I believe to start the countdown
> 
> Hmm, perhaps I should try that. ;-)

No joy.

**Comment by JeffAbrahamson** [comment]
I'll have a closer look next week.

**Comment by mcmikashi** [comment]
Change : 

- Small fix : check if the url have get data if not add a new get data because the list url when we create a new item doesn't have the “from” in the url.

For delete tb_oject we need to check the list_items_by_slug view once we do that if he doesn't have cheduled_for_deletion_date it will set to today's date and after 15 days it will be deletable with the delete url.

**Comment by JeffAbrahamson** [comment]
Hmm, clicking the trash can icon (which I think is meant to clean up even though the icon suggests deletion) adds a `delete=true`.  Each time.  So here I am at

`http://localhost:8000/tb/admin/t/list/une-meilleure-mobilit-pour-le-climat/?from=articles&delete=true&delete=true`

**Comment by JeffAbrahamson** [comment]
I've fixed the conflicts and merged this.

**Pull request closed without merging**